.github/workflows/ci.yml

```diff
@@ -53,20 +53,14 @@ jobs:
           python -m pip install gcovr
           gcovr -r . --xml-pretty --output coverage-cpp.xml
       
-      - name: Merge Coverage Reports
-        run: |
-          python -m pip install coverage
-          coverage combine coverage-python.xml coverage-cpp.xml
-          coverage xml -o merged-coverage.xml
-      
       - name: Upload Coverage
         uses: codecov/codecov-action@v3
         with:
           token: ${{ secrets.CODECOV_TOKEN }}
           directory: ./coverage/reports/
           env_vars: OS,PYTHON
           fail_ci_if_error: false
-          files: ./merged-coverage.xml
+          files: ./coverage-python.xml,./coverage-cpp.xml
```

Comment:
This works?

Reply:
I have arrived at a preliminary solution for merging Python and C++ coverage reports using Codecov, based on their official documentation https://docs.codecov.com/docs/github-4a-merging-reports.

To further validate this solution, I plan to develop and test it locally by utilizing a Codecov API key. I will ensure comprehensive testing and provide you with an update on the results by the end of the week sir .

Reply:
Actually there should be a way to do it without Codecov. This is indeed tough to achieve. So if you want to take other issues while parallelly working on it.